vermouth/tests/test_molecule.py

```diff
@@ -132,3 +140,610 @@ def test_link_predicate_match():
+    Test that LinkParameterEffector compare different if they have different keys.
+    """
+    n_keys = effector_class.n_keys_asked
+    left_keys = ['A{}'.format(idx) for idx in range(n_keys)]
+    right_keys = ['B{}'.format(idx) for idx in range(n_keys)]
+    left = effector_class(left_keys)
+    right = effector_class(right_keys)
+    assert left != right
+
+
+@pytest.mark.parametrize('left_class, right_class', itertools.combinations((
+    vermouth.molecule.ParamDistance,
+    vermouth.molecule.ParamAngle,
+    vermouth.molecule.ParamDihedral,
+    vermouth.molecule.ParamDihedralPhase,
+), 2))
+def test_link_parameter_effector_diff_class(left_class, right_class):
+    """
+    Test that LinkParameterEffector compare different if they have different classes.
+    """
+    left_n_keys = left_class.n_keys_asked
+    left_keys = ['A{}'.format(idx) for idx in range(left_n_keys)]
+    left = left_class(left_keys)
+
+    right_n_keys = right_class.n_keys_asked
+    right_keys = ['B{}'.format(idx) for idx in range(right_n_keys)]
+    right = right_class(right_keys)
+
+    assert left != right
+
+
+@st.composite
+def random_interaction(draw, graph, natoms=None,
+                       interaction_class=Interaction, attrs=False):
+    if natoms is None:
+        natoms = draw(st.integers(min_value=0, max_value=len(graph.nodes)))
+    # The test is only relevant with a user-provided value of natoms.
+    hypothesis.assume(0 <= natoms <= len(graph.nodes))
+    atoms = tuple(draw(st.sampled_from(list(graph.nodes))) for _ in range(natoms))
+    # TODO: Allow for LinkParameterEffector instances.
+    parameters = st.lists(elements=st.text())
+    # TODO: Allow for more complex meta attributes.
+    meta = draw(st.one_of(st.none(), st.fixed_dictionaries({})))
+    if attrs:
+        atom_attrs = tuple(draw(st.fixed_dictionaries({})) for _ in atoms)
+        return interaction_class(
+            atoms=atoms,
+            atom_attrs=atom_attrs,
+            parameters=parameters,
+            meta=meta,
+        )
+    return interaction_class(atoms=atoms, parameters=parameters, meta=meta)
+
+
+@st.composite
+def interaction_collection(draw, graph,
+                           interaction_class=Interaction, attrs=False):
+    result = {}
+    ninteraction_types = draw(st.integers(min_value=0, max_value=10))
+    for _ in range(ninteraction_types):
+        ninteractions = draw(st.integers(min_value=0, max_value=10))
+        type_name = draw(st.text())
+        if type_name not in result and ninteractions > 0:
+            result[type_name] = []
+        for _ in range(ninteractions):
+            interaction = draw(random_interaction(
+                graph, interaction_class=interaction_class, attrs=attrs,
+            ))
+            result[type_name].append(interaction)
+    return result
+
+
+@st.composite
+def random_molecule(draw, molecule_class=Molecule):
+    # TODO: Allow for more complex atom attributes.
+    graph = draw(hnst.graph_builder())
```

Comment:
You can also pass molecule_class to the graph_builder with keyword `graph_type`.